internal/venice-common/src/main/java/com/linkedin/venice/pubsub/manager/TopicMetadataFetcher.java

```diff
@@ -183,9 +183,14 @@ void validateTopicPartition(PubSubTopicPartition pubSubTopicPartition) {
     if (pubSubTopicPartition.getPartitionNumber() < 0) {
       throw new IllegalArgumentException("Invalid partition number: " + pubSubTopicPartition.getPartitionNumber());
     }
-    if (containsTopicCached(pubSubTopicPartition.getPubSubTopic())) {
-      return;
+    try {
+      if (containsTopicCached(pubSubTopicPartition.getPubSubTopic())) {
+        return;
+      }
+    } catch (Exception e) {
+      // in case there is any exception, we will retry the operation
```

Comment:
Shouldn't we classify what is recoverable vs irrecoverable exceptions here to deem that we can succeed in retrying the same flow? Is that possible?

Reply:
Yeah we already have that distinction. Will update

Reply:
Done